Add published date to PirateBay results

ducalex opened the pull request:
This PR adds the date field to the results from the piratebay.py plugin.

It is meant to go with the recently merged PR: https://github.com/qbittorrent/qBittorrent/pull/20703

The change does _not_ break compatibility with older (current) qBittorrent versions.

![image](https://github.com/qbittorrent/search-plugins/assets/13711380/0ad4ab51-f0b2-4b9a-9115-d490f7de4abe)

xavier2k6 commented:
@ducalex same thing that came up from that PR in main repo, should commit be `Added published date column`? , also `versions.txt` needs to be bumped.

ducalex commented:
I've addressed the issues you mentioned!

When my date PR was merged it was indeed Published Date but I don't know if it will change by the time it gets to a release (I think it was quickly merged by maintainers to avoid bikeshedding on the name). So feel free to re-title my commit before merging as you see fit.

xavier2k6 commented:
@ducalex Thank You!
